### PR TITLE
`ictr_score_companies()` now errors if `inputs_co2` has `NA`s

### LINKED ISSUE #149: Test to check NAs in carbon footprint column

### PR DESCRIPTION
**Pull request by kalashsinghal**
closes #149 

Dear @maurolepore 

This test is subject to change if we have to handle NAs in the real data. Currently, I made a test which make sure that we don't have any NA value in the input_co2 column. Please feel free to suggest me if a better test case could be written for this issue! Thanks!

FYI @Tilmon

### CODE REVIEW

File: tests/testthat/test-ictr_score_companies.R [hunk 1]
```diff
@@ -74,3 +74,7 @@ test_that("returns 3 rows per company for any slice of inputs", {
   out <- ictr_score_companies(data, two_companies)
   expect_equal(nrow(out), 6L)
 })
+
+test_that("There are no NAs in the input_co2 column", {
+  expect_false(any(is.na(ictr_inputs$input_co2)))
+})
```
Comment (maurolepore) [inline tests/testthat/test-ictr_score_companies.R:80]: Thanks Kalash for considering this edge case and drafting the test. 

I think this particular test is testing not the real data but the example data we have in the tiltIndicator package. If that's what you want to test, then the file for this test would be `tests/testthat/test-ictr_data.R.

But what may be interesting to test is that our functions handle missing values gracefully -- for example by throwing an error alerting the user. Then they'll need to remove the `NA` before they can use the function.

```
library(testthat)
library(tibble)

# Comes with tiltIndicator
demo_data <- tibble(input_co2 = 1)

# tests/testthat/test-demo_data.R
test_that("our `demo_data` has no missing value in `input_co2`", {
  expect_false(anyNA(demo_data$input_co2))
})

# R/f.R
f <- function(data) {
  if (anyNA(data$input_co2)) stop("Each `input_co2` must be not missing.")
  
  # more code
}

# tests/testthat/test-f.R
test_that("with a missing value of `input_co2` errors gracefully", {
  data <- tibble(input_co2 = c(1, NA))
  expect_error(f(data), "input_co2.*missing")
})
```


``` r
rank(c(1, 2, NA, 3, 4), na.last = TRUE)
#> [1] 1 2 5 3 4
rank(c(1, 2, NA, 3, 4), na.last = FALSE)
#> [1] 2 3 1 4 5
```

**Comment by maurolepore** [comment]
Ups, I'm on my phone sorry I accidentally marked as ready for review 

Can't figure out how to revert it.

File: tests/testthat/test-ictr_score_companies.R (hunk 1 above)
Comment (maurolepore) [reply tests/testthat/test-ictr_score_companies.R]: I like your idea of creating a separate issue so you can merge this PR and continue the conversation there. Do engage Tilman and maybe show a reprex to explain how this would change the output.

**Review by maurolepore — APPROVED** [review]
LGTM. Note I merged the PR that you reviewed before into this one.

I like your idea of opening a new issue about how to handle `NA`s in `rank()` so we can show a reprex to Tilman and see what behaviour we wants.
